core/src/main/scala/filodb.core/store/ChunkSetInfo.scala

```diff
@@ -129,18 +130,26 @@ object ChunkSetInfo extends StrictLogging {
   /**
    * ChunkSetInfo metadata schema:
    * +0  long   chunkID
-   * +8  int    # rows in chunkset
-   * +12 long   start timestamp
-   * +20 long   end timestamp
-   * +28 long[] pointers to each vector
+   * +8  long   ingestion time (milliseconds from 1970)
+   * +16 int    # rows in chunkset
+   * +20 long   start timestamp
+   * +28 long   end timestamp
+   * +36 long[] pointers to each vector
    *
    * Note that block metadata has a 4-byte partition ID appended to the front also.
+   *
+   * Note 2: The chunkID is currently defined from the start timestamp and the ingestion time,
+   * and so it can be derived instead of being stored explicitly. The ChunkMap requires that
+   * the chunkID be readily available at offset 0, which is one reason to have it. It's
+   * possible that ChunkMap could call a method instead, but storing the chunkID also offers
+   * flexibility in the future, in case the definition changes.
    */
   val OffsetChunkID = 0
-  val OffsetNumRows = 8
-  val OffsetStartTime = 12
-  val OffsetEndTime = 20
-  val OffsetVectors = 28
+  val OffsetIngestionTime = 8
```

Comment:
Why is IngestionTime needed in the ChunkInfo ?
Is it useful at all long term ?

Reply:
Two reasons: 1. I cannot easily stash the ingestion time anywhere else and pass it along down to the Cassandra layer. The full ingestion time cannot be derived from the chunk ID.
2. It's always a good idea to have all the information necessary to reconstruct a secondary index from the primary table. Without this, this the full ingestion time wouldn't be stored in the main chunk table.